### ISSUE TITLE
Docker issues

### ISSUE THREAD

**Issue by PySimpleSQL** [PySimpleSQL/pysimplesql#225]
Just creating a separate ticket to track Docker issues.

**Comment by PySimpleSQL** [comment]
I have some cleanup to do next. Currently this is a bit hard-coded for Postgres; I need to generalize a few things so that I can create Docker images for other databases as well

**Comment by PySimpleSQL** [comment]
Some more notes for testing and identifying the "first connection fails" issue:


From a terminal run:
```
docker rm -f $(docker ps -a -q)
```
To force the next run of journal_postgres_docker.py to rebuild the container

If you also want to force a re-download of the Docker image, also run this command from a terminal after the above command:
```
docker rmi $(docker images -a -q)
```

if you want to see the logs from the Docker container for troubleshooting, run the following from a terminal:
```
docker logs pysimplesql-examples-postgres
```

Notice in journal_postgres_docker.py the first connection attempt fails each time it is run.  If you comment out 
```
docker_container.stop()
```
in the main loop exit event, then the container remains running and only the very first connection attempt after container start fails.

This is the last bug to fix and the Docker testing integration is done.

**Comment by PySimpleSQL** [comment]
Creating the Docker Image:

Run the Docker_create.py file in examples/PostgreSQL_examples/docker
It will ask for a username and a password:
username:pysimplesql
password:<ssweber -- use the password I gave you privately>

This downloads the builds a new image from the Dockerfile (Dockerfile2 currently), starts a container to populate the database, saves the container as a new image and pushes it to DockerHub using the username/password entered.

This really only needs run if we ever want to change the base Docker imageand the data inside - so really, it may never need run again.  Just wanted to show how the base pysimplesql/examples:postgres image is being created.

**Comment by ssweber** [comment]
Tested all three on my laptop. Sqlserver and Postgres worked, but had the connection issue with MySQL.

 When I closed gui, and tried again it worked immediately. When I get back on my laptop I’ll post the logs.

**Comment by PySimpleSQL** [comment]
Thanks.  I believe that the same happened for me on Linux.  I'll track it down soon - it may just be a timing issue for the database to initialize.  Tonight is a light one for me - it's a pretty steady night at work, so I won't be able to get into anything too crazy

I've been super impressed by Docker.  How nice to be able to set up a server for testing - and for end users trying examples. They can test while having absolutely no knowledge of setting up a server at all.  I'm going to start using Docker more on my development machine.  It's so much cleaner than riddling your whole system with a bunch of servers that you don't  need running all the time.

**Comment by ssweber** [comment]
Totally. We’ll probably want to take down the heliohost examples to avoid vandalism, unless you’ve set it up to refresh the data daily.

On a different note, do you have the docker file and helper script for the sqlserver? I saw you uploaded the MySQL/Postgres ones, but it’s was missing.

**Comment by PySimpleSQL** [comment]
I did the Docker stuff manually on the command line for the SQLServer
image - I do plan on following up with a script but just didn't have time
yet. Plus I was using it as a learning experience.

I've also been thinking about the HelioHost versions.  I hate that they are
so slow, but I like that it makes testing an example super easy for a user
(there is no Docker setup required).  I do have pysimplesql_user set with
very limited permissions to keep vandalism to a minimum, but of course it's
pretty hard to eliminate it entirely.  I could definitely see if I can have
a script on the server that wipes it daily, that's not a bad idea at all.

**Comment by ssweber** [comment]
Fixed the docker issues #338
